clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1513,9 +1530,38 @@ protected void processCommonConsumerAction(
           ProducerTracker producerTracker = kafkaDataIntegrityValidator.registerProducer(producerGuid);
           producerTracker.setPartitionState(partition, entry.getValue());
         });
-
         long consumptionStatePrepTimeStart = System.currentTimeMillis();
         checkConsumptionStateWhenStart(offsetRecord, newPartitionConsumptionState);
+
+        if (newPartitionConsumptionState.isRestartIngestion()) {
+          // remove the states set above
+          partitionConsumptionStateMap.remove(partition);
+          kafkaDataIntegrityValidator.clearPartition(partition);
```

Comment:
In the unsub flow, we do both of these, and also `storageUtilizationManager.removePartition(partition);`... would it be supposed to happen here as well? This looks like it may benefit from having a separate function to do all of the things we need, and to call it from both places. WDYT?

Reply:
Thanks for pointing this out. Found the `RESET_OFFSET` flow more apt here, reusing it 🚀